### PR TITLE
UserMainMenuImpl

### PR DESCRIPTION
**Pull request by tbondare**

### CODE REVIEW

**Review by KrasuliaV — COMMENTED** [review]
You create a scanner in almost every place. May you create it in one place and use it everywhere?

File: src/main/java/view/impl/UserMainMenuImpl.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                exit();
+            default :
+                System.out.println("Wrong number. Please, enter correct number");
+                this.productChooserMenu();
+        }
+    }
+
+    @Override
+    void ordersMenuShow(User user) {
+        ordersShow(user);
```
Comment (KrasuliaV) [inline src/main/java/view/impl/UserMainMenuImpl.java:42]: Why do you create one additional method?

File: src/main/java/view/impl/UserMainMenuImpl.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        }
+        printMessage("choose products");
+        answer = scanAnswer(this::productsShow);
+        if (answer.equals("1")) {
+            productChooserMenu();
+        }
+    }
+
+    void ordersShow(User user) {
+        ArrayList<Order> orders = user.getOrderList();
```
Comment (KrasuliaV) [inline src/main/java/view/impl/UserMainMenuImpl.java:62]: What user must do when the order is emty?

File: src/main/java/view/impl/UserMainMenuImpl.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+//        если нет
+//          ретерн назад;
+    }
+
+    void exit();
+    {
+        new LoginMenu.show();
+    }
+
+    private void checkOrderForExistence() {
```
Comment (KrasuliaV) [inline src/main/java/view/impl/UserMainMenuImpl.java:138]: Where do you use this method?

File: src/main/java/view/impl/UserMainMenuImpl.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    void ordersShow(User user) {
+        ArrayList<Order> orders = user.getOrderList();
+        for (Order currentOrder : orders) {
+            System.out.println(currentOrder.toString());
+        }
+        printMessage("checkout order");
+        answer = scanAnswer(() -> ordersShow(user));
+        if (answer.equals("1")) {
+            checkoutOrder();
```
Comment (KrasuliaV) [inline src/main/java/view/impl/UserMainMenuImpl.java]: Which order should the user checkout?

File: src/main/java/view/impl/UserMainMenuImpl.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    private String scanAnswer(Runnable method) {
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine();
+        if (input == "1") {
+            return "1";
+        }
+        if (input == "2") {
+            showSubMenu(SubMenu submenu, User user);
```
Comment (KrasuliaV) [inline src/main/java/view/impl/UserMainMenuImpl.java:153]: May you explain please what is it?

**Review by Satinari — APPROVED** [review]

**Review by KrasuliaV — APPROVED** [review]

**Pull request merged**